### DISCUSSION TITLE
Swagger doesn't display properly when using query param model and other query params

### DISCUSSION THREAD

**Discussion by riosc7** [fastapi/fastapi#12535] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Annotated, Literal

from fastapi import FastAPI, Query
from pydantic import BaseModel, Field

app = FastAPI()


class FilterParams(BaseModel):
    limit: int = Field(100, gt=0, le=100)
    offset: int = Field(0, ge=0)
    order_by: Literal["created_at", "updated_at"] = "created_at"
    tags: list[str] = []


@app.get("/items/")
async def read_items(other: Annotated[int, Query()], filter_query: Annotated[FilterParams, Query()]):
    return filter_query
```

### Description

When I display the `docs` I expect a swagger with the defined endpoint showing all declared query params independently. Instead, I get a swagger showing the first query params and all other query params in an object, like a body, all together. This doesn't happen if the endpoint is defined only with the `FilterParams` models, as the example in documentation.
![image](https://github.com/user-attachments/assets/a02c03ae-b7c7-4c25-a7fe-d500e04dd5d1)

### Operating System

Linux

### FastAPI Version

0.115.0

### Pydantic Version

2.9.2

### Python Version

Python 3.11.9

**Comment by YuriiMotov** [comment]
Duplicate of https://github.com/fastapi/fastapi/discussions/12212.
To keep the discussions clean, let's close this topic and track it there.
